buildSrc/src/main/scala/mesosphere/gradle/protobuf/ScalaPb.scala

```diff
@@ -0,0 +1,59 @@
+package mesosphere.gradle.protobuf
+
+import java.io.File
+import java.util.concurrent.atomic.AtomicInteger
+
+import com.typesafe.scalalogging.StrictLogging
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.FileTree
+import org.gradle.api.tasks.TaskAction
+import org.gradle.process.ExecResult
+
+import scala.beans.BeanProperty
+import collection.JavaConverters._
+
+/**
+  * Gradle task that runs ScalaPb protobuf code generation.
+  *
+  * @see https://scalapb.github.io/scalapbc.html
+  * @see https://github.com/etiennestuder/gradle-jooq-plugin/
+  * @see https://github.com/google/protobuf-gradle-plugin
```

Comment:
I think we really should model it after the Google Protobuf plugin and go with something like

```groovy
project(':persistence-zookeeper') {
  apply plugin: 'com.mesosphere.protobuf'
}
```

this should then create a `proto` source set.